x-pack/plugins/lens/public/xy_visualization/visualization.tsx

```diff
@@ -300,6 +302,20 @@ export const getXyVisualization = ({
     return getFirstDataLayer(state.layers)?.palette;
   },
 
+  onDrop(props) {
+    const targetLayer: XYLayerConfig | undefined = props.prevState.layers.find(
+      (l) => l.layerId === props.target.layerId
+    );
+    if (!targetLayer) {
+      return props.prevState;
+    }
```

Comment:
You've articulated the dilemma well. Thinking out loud here.

When we make code robust to situations that _should_ never occur it sometimes creates harder-to-find bugs than if we just let the code obviously crash and burn. 

On the other hand, as a user, I don't want a minor (possibly inconsistent) bug to become so disruptive that the application is no longer useful (or worse, that I lose my work).

Thinking about this drag-and-drop case, I agree that it seems like the handling should be bullet-proof because, like you say, the impossible _could_ happen and the best thing for the user is probably to shove it under the rug.

If I'm dealing with a different instance of the same pattern you described I might feel differently. For example, if a visualization I'm expecting to be in the `visualizationMap` isn't present, that's a big enough issue that I think the best thing for the user is to make the problem obvious so that we catch it early before the release.

And maybe our release cadence and development process inform these decisions, too. For example, maybe obvious bugs become more attractive with a slow release schedule because we have a long time to catch things before actually putting software in front of our users.

Someone must have written a book about this. It could be a good discussion to have as a team.

But, I think how you're doing it in this case makes sense. 👍